app/main.py

```diff
@@ -1,34 +1,56 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.create_decks()
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def create_decks(self) -> None:
+        self.decks = []
+        for row in range(self.start[0], self.end[0] + 1):
+            for column in range(self.start[1], self.end[1] + 1):
+                self.decks.append(Deck(row, column))
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if (deck.row, deck.column) == (row, column):
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+
+        if not any([deck.is_alive for deck in self.decks]):
```

Comment:
```suggestion
        if not any(deck.is_alive for deck in self.decks):
```

Remove redundant `[]`